src/utils/index.js

```diff
@@ -32,15 +32,20 @@ const isValidDate = (dateStr) => {
   return moment(dateStr) && date.isValid();
 };
 
-const getCourseUrlSlugPattern = (updatedSlugFlag, courseRunStatuses, productSource) => {
+const getCourseUrlSlugPattern = (updatedSlugFlag, courseRunStatuses, productSource, CourseType) => {
   /**
   * This function returns the course url slug pattern based on the new slug enabled flag and courseRunStatuses
   */
-  if (updatedSlugFlag && productSource === DEFAULT_PRODUCT_SOURCE && courseRunStatuses.some((status) =>
-    // eslint-disable-next-line implicit-arrow-linebreak
-    (IN_REVIEW_STATUS.includes(status) || POST_REVIEW_STATUSES.includes(status)))) {
-    // change to COURSE_URL_SLUG_PATTERN_NEW when rollout is complete
-    return COURSE_URL_SLUG_PATTERN;
+  if (updatedSlugFlag) {
+    if (productSource === DEFAULT_PRODUCT_SOURCE && courseRunStatuses.some((status) =>
+      // eslint-disable-next-line implicit-arrow-linebreak
+      (IN_REVIEW_STATUS.includes(status) || POST_REVIEW_STATUSES.includes(status)))) {
+      // change to COURSE_URL_SLUG_PATTERN_NEW when rollout is complete
+      return COURSE_URL_SLUG_PATTERN;
+    }
+    if (CourseType === EXECUTIVE_EDUCATION_SLUG && productSource === DEFAULT_EXTERNAL_PRODUCT_SOURCE) {
+      return EXEC_ED_COURSE_URL_SLUG_PATTERN;
+    }
```

Comment:
Again, we can utilize env variables in edx-internal to achieve this.

Reply:
I'm refactoring the code using dict. I will move it to `edx-internal` after the changes